android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java

```diff
@@ -112,6 +117,21 @@ public void onStartInput(EditorInfo attribute, boolean restarting) {
       KMManager.setNumericLayer(KeyboardType.KEYBOARD_TYPE_SYSTEM);
     }
 
+    // Temporarily disable predictions if entering a hidden password field
+    if (KMManager.isHiddenPasswordInputType(inputType)) {
```

Comment:
I am a little concerned that this has the side effect of setting a global state variable (`shouldOverrideMayProtect`) which is not great for maintenance in the future. Is it possible to either (a) rename, or (b) preferably, refactor this so no global variable is required?

Reply:
I've separated the getter / setter of the private state variable.